validator_client/src/duties_service.rs

```diff
@@ -214,6 +214,8 @@ pub struct DutiesService<T, E: EthSpec> {
     pub sync_duties: SyncDutiesMap<E>,
     /// Provides the canonical list of locally-managed validators.
     pub validator_store: Arc<ValidatorStore<T, E>>,
+    /// Maps unknown validator pubkeys to the next slot time when a poll should be conducted again.
+    pub unknown_validator_next_poll_slots: RwLock<HashMap<PublicKeyBytes, Slot>>,
```

Comment:
Does this need to be behind a lock? Not that big of an overhead but seems like its just the update service that uses it.

Reply:
Not sure to be honest - I initially tried using `HashMap` directly but ran into some issues. The async task that runs the polling only gets a immutable reference to the `DutiesService` via an `Arc`, and I couldn't borrow it as a mutable directly,  so I kinda borrowed the existing pattern that's used to mutate `DutiesService`. Would be happy to get rid of this if there's a way to do this nicely.

Reply:
Makes sense, since we pass around a reference, I think this is the only clean way to mutate.